libtenzir/include/tenzir/table_slice.hpp

```diff
@@ -402,6 +402,9 @@ auto resolve_operand(const table_slice& slice, const operand& op)
 auto unflatten(const table_slice& slice,
                std::string_view nested_field_separator) -> table_slice;
 
+/// Just like `unflatten`, but... better? worse?
+auto unflatten2(const table_slice& slice, std::string_view sep) -> table_slice;
```

Comment:
As much as I laughed about the joke, I think we should actually describe the difference here and maybe even choose to get rid of the old implementation. I think the new one better.